internal/terraform/eval_context_builtin.go

```diff
@@ -426,6 +426,26 @@ func (ctx *BuiltinEvalContext) EvaluateReplaceTriggeredBy(expr hcl.Expression, r
 		return nil, false, diags
 	}
 
+	// Validate the attribute reference against the target resource's schema.
+	// We use schema-based validation rather than value-based validation because
+	// resources may contain dynamically-typed attributes (DynamicPseudoType) whose
+	// actual type can change between plans. Schema validation ensures we only
+	// error on truly invalid attribute references.
+	providerAddr := ctx.Evaluator.Config.ResolveAbsProviderAddr(resCfg.ProviderConfigAddr(), ctx.Path().Module())
```

Comment:
The provider configuration is not necessarily local to the current module, we need to resolve the correct provider instance, since the local name may not match the provider name. It's kind of awkward to lookup, because the context isn't supposed to know how providers are mapped, it's told what provider to use in absolute terms.

In this case we have the provider information already in the change, because that is how we encode/decode it.
```
providerSchema, err := ctx.ProviderSchema(change.ProviderAddr)
```